cli/src/main.rs

```diff
@@ -16,37 +22,58 @@ mod tracker;
 use anyhow::{bail, Error};
 use checksum_records::ChecksumRecords;
 use clap::Parser;
-use cli_args::{CargoCli, Command};
+use cli_args::{CargoCli, Command, UiMode};
+use console::style;
 use dev_environment::{print_env_info, resolve_dev_tools};
+use fail_fast::set_fail_fast;
 use job_type::JobType;
+use jobs_runner::cancellation::{cancellation_token, graceful_shutdown};
 use location::init_location;
-use spawner::SpawnResult;
-use std::{
-    fs::File,
-    io::{self, stdout, Stdout},
-    path::PathBuf,
-};
+use log_print::{print_log_separator, print_report};
+use release::do_release;
 use target::Target;
-use tracker::get_tracker;
+use tokio::signal;
+use tracker::{get_tracker, init_tracker};
 
 use crate::cli_args::EnvironmentCommand;
 
-#[derive(Debug)]
-pub enum ReportOptions {
-    None,
-    Stdout(Stdout),
-    File(PathBuf, File),
-}
-
 #[tokio::main]
 async fn main() -> Result<(), Error> {
+    // CLI command parsing.
     let CargoCli::Chipmunk(cli) = CargoCli::parse();
+    let command = cli.command;
 
+    // Validate current directory location.
     init_location()?;
 
+    // Handle the app main process in a separate method, keeping this method for handling
+    // manual cancelling as well.
+
+    tokio::select! {
+        main_res = main_process(command) => {
```

Comment:
Does it make sense to use here `join` as soon as you are canceling both hands in any way? With `select` you can probably miss a fact of incorrect behavior of `main_process`, because `main_process` will be shutdown with `select` in any way. With `join` you have to wait reaction of `main_process` hand and it will allow notice possible issues with shutdown.